### PR TITLE
nlplug-findfs: enable automated testing

### PR DESCRIPTION
**Pull request by 7heo**
Re-opened for Travis testing.

### CODE REVIEW

**Comment by 7heo** [comment]
Fails with a segfault, as expected. Can be safely merged, the segfault fix follows.

**Comment by jirutka** [comment]
Why is this not merged yet?

**Comment by 7heo** [comment]
I'll tell you after sunset. :)

**Comment by Klowner** [comment]
23+ sunsets

**Comment by 7heo** [comment]
@Klowner I know... I lag. Sorry. I'll do it asap. Turns out I expected to have time that evening after sunset (not going out much at night)... and I didn't. Again, I'll do it asap. :)

**Comment by Klowner** [comment]
heheh, it's okay, just poking you

**Comment by 7heo** [comment]
Thanks for poking. Eventually, I'll be poked when I have free time and do it ;)

**Comment by 7heo** [comment]
So, @jirutka to answer your question after a 58 sunsets delay, this hasn't been merged because:

1. I have to make sure that the mkinitfs tree and history is correct (with the right commits in the right order)
2. I have to make sure that the mkinitfs binary preforms as expected in this commit.

File: nlplug-findfs.c [hunk 1]
```diff
@@ -528,9 +532,13 @@ static int read_pass(char *pass, size_t pass_size)
 	}
 	pass[strlen(pass) - 1] = '\0';
 
-	if (tcsetattr(STDIN_FILENO, TCSANOW, &old_flags) < 0) {
-		warn("tcsetattr");
-		return r;
+	if (isatty(fileno(stdin))) {
+		if (tcsetattr(STDIN_FILENO, TCSANOW, &old_flags) < 0) {
```
Comment (ncopa) [inline nlplug-findfs.c:536]: how about: `if (isatty(STDIN_FILENO) && (tcsetattr(STDIN_FILENO, TCSANOW, &old_flags) < 0) {`

File: nlplug-findfs.c [hunk 2]
```diff
@@ -516,10 +516,14 @@ static int read_pass(char *pass, size_t pass_size)
 	new_flags.c_lflag &= ~ECHO;
 	new_flags.c_lflag |= ECHONL;
 
-	r = tcsetattr(STDIN_FILENO, TCSANOW, &new_flags);
-	if (r < 0) {
-		warn("tcsetattr");
-		return r;
+	if (isatty(fileno(stdin))) {
```
Comment (ncopa) [inline nlplug-findfs.c]: use `STDIN_FILENO` instead of `fileno(stdin)`

File: nlplug-findfs.c (hunk 1 above)
Comment (7heo) [reply nlplug-findfs.c:536]: It is not the same logical computation.

**Comment by jirutka** [comment]
Merged in 47e87b829c09ae3c6f3bcbabc9824d4be9bf35f9 after small clean-up. Thanks!

**Pull request closed without merging**